src/components/UploadFiles.tsx

```diff
@@ -27,29 +35,28 @@ export const UploadFiles: FC<UploadFilesProps>  = ({client,bucket,setKey, canUpl
     }
     const [files, setFiles] = useState<File[]>([]);
     const handleUpload = async () => {
-        const compressedFiles: File[] = [];
-        if (files) {
-            for (let i = 0; i < files.length; i++) {
-                const element = files[i];
+        let filesToUpload: File[];
+        let key = '';
 
-                // Read the file content as a Buffer
-                const compressedFile = await compressFile({element,element_name: element.name})
+        if (compressible)
+            filesToUpload = await Promise.all(files.map(async (file) => {
+                return await compressFile({element: file, element_name: file.name})
+            }))
+        else
+            filesToUpload = files;
 
-                compressedFiles.push(compressedFile);
-            }
-        }
+        if (filesToUpload) {
+            filesToUpload.forEach(fileToUpload => {
+                // assign a unique name for the file, usually has to timestamp prefix
+                key = `${Date.now()}__${fileToUpload.name}`
```

Comment:
Shouldn't this be a default value if the `key` prop is undefined ?

Reply:
if so, we are not using said prop

Reply:
the value of the key comes from fileToUpload and if its undefined we can't make all the upload process